### DISCUSSION TITLE
다른 Module 의 Respository 를 사용해도 될까요?

### DISCUSSION THREAD

**Discussion by jis-kim** [GhostPangPang/GhostPong#48] [Q&A] [answered]
friend API 구현 중에 User repository 쓸 일이 좀 있습니당. 지금 UserService 에 이런저런 함수가 있을 것이다 ( ex - 닉네임으로 id 찾는 함수) 라고 가정하고 하고 있는데 이게 맞는지 아니면 다른 Module 에서도 Repository 를 그냥 쓰는 게 맞는 건지 모르겠어요…

**Comment by jis-kim** [comment]
소현) 아ㅏ 이거는 초짜인 저도 좀 고민되는 부분이긴 한데 repository를 가져다 쓰는게 객체지향적인건지 service에서 가져다 쓰는게 좋은 건지 잘 모르겠긴 해요….

근데 friend에서 user에 요청할 것들이 모두 service에 있음이 확실하고 service만 불러오면 다 해결할 수 있는거면 몰라도 service불러오고 다른 로직 때문에 repository를 또 불러올거면 그냥 repository만 불러서 원하는 로직 처리하는게 덜 결합되는거 아닌가 싶기도 하고

아니면 걍 repository를 아무나 못부르게 하는게 좋은건가 싶기도 하고….

글로 쓴 제 말 이해가시나요??? ㅋㅋㅋㅋㅋ이따가 구두로 한번 더 말해보겠습니다.ㅎㅎㅎ
이런건 클린코드나 뭐 아키텍처 이런데 좀 가이드가 나와있지 않을까 싶기도 하네요… 움… 어렵네요.ㅠㅜ

**Comment by anso33** [comment, marked as answer]
음 근데 방금 스프링 인강 들었던 코드 봤는데 보면 또 그냥 서비스에서 다른 도메인 repository를 가져다 썼더라고요? 이건 팀플이 아니니 우리가 우려했던 상황같은건 고려하지 않았어도 될텐데 말이죠?ㅋㅋㅋㅋㅋㅋㅋㅋ음 이건 백엔드 경험이 있는 사람들에게 좀 물어보고 오겠습니다. 알면 좋으니까~~~

**Comment by jis-kim** [comment]
오노 ㅋㅋㅋㅋㅋㅋ 일단 Respository 각각 쓰고 REST 합치고 생각해봅시다!
